src/crystal/system/unix/sysctl_cpucount.cr

```diff
@@ -0,0 +1,17 @@
+{% skip_file unless flag?(:openbsd) || flag?(:freebsd) %}
+
+require "c/sysctl"
+
+module Crystal::System
+  def self.cpu_count
+    mib = Int32[LibC::CTL_HW, LibC::HW_NCPU]
+    ncpus = 0
+    size = sizeof(Int32).to_u64
+
+    if LibC.sysctl(mib, 2, pointerof(ncpus), pointerof(size), nil, 0) == 0
+      ncpus
+    else
+      -1
```

Comment:
Possibly? I was just matching the behaviour of the existing `sysconf` version 

Is that the behaviour of other standard Crystal libs, or do they just rely on return value?